internal/webserver/handler-api.go

```diff
@@ -278,6 +286,14 @@ func (h *handler) apiInsertBookmark(w http.ResponseWriter, r *http.Request, ps h
 		}
 	}
 
+	// Use the user-supplied title and excerpt if they exist
+	if userExcerpt != "" {
+		book.Excerpt = userExcerpt
+	}
+	if userTitle != "" {
+		book.Title = userTitle
```

Comment:
the `core.ProcessRequest` do have a `KeepTitle` attr to keep the title when proccessing the bookmark. Maybe this can be implemented by `KeepTitle:   book.Title != ""` in line 278

Reply:
This is a cleaner way to implement this :)